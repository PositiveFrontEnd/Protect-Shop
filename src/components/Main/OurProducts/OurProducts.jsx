import React from "react";
import DesktopProducts from "./Desktop/DesktopProducts";
import MobileProducts from "./Mobile/MobileProducts";
import { useState, useEffect } from "react";

const OurProducts = () => {
  const [isMobile, setIsMobile] = useState(
    window.matchMedia("(max-width: 768px)").matches
  );
  const handleMediaQuery = (event) => {
    setIsMobile(event.matches);
  };
  useEffect(() => {
    const mediaQueryList = window.matchMedia("(max-width: 768px)");
    mediaQueryList.addListener(handleMediaQuery);
    return () => {
      mediaQueryList.removeListener(handleMediaQuery);
    };
  }, []);
  return <>{isMobile ? <MobileProducts /> : <DesktopProducts />}</>;
};

export default OurProducts;
