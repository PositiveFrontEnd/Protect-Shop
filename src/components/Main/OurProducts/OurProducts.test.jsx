import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import OurProducts from "./OurProducts";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

describe("Тестування компонента  OurProducts", () => {
    test("Знімок", () => {
        const mockStore = configureStore([thunk]);
        const initialState = {

        };

        const store = mockStore(initialState);

        const ourProducts = render(
            <Provider store={store}>
                <BrowserRouter>
                    <OurProducts />
                </BrowserRouter>
            </Provider>
        );

        expect(ourProducts).toMatchSnapshot();
    });
});
