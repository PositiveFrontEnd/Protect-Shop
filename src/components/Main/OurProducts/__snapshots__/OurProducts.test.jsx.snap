// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Тестування компонента  OurProducts > Знімок 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div
          class="products__wrapper"
        >
          <div
            class="container products"
          >
            <div
              class="products__images"
            >
              <div
                class="products__images"
              >
                <img
                  alt=""
                  src="/Images/forWomen.jpg"
                />
                <a
                  class="button-black products__image__caption"
                  href="/catalogue?categories=woman"
                >
                  for woman
                </a>
              </div>
              <div
                class="products__images"
              >
                <img
                  alt=""
                  src="/Images/forKids.jpg"
                />
                <a
                  class="button-black products__image__caption"
                  href="/catalogue?categories=children"
                >
                  for woman
                </a>
              </div>
              <div
                class="products__images"
              >
                <img
                  alt=""
                  src="/Images/forMen.jpg"
                />
                <a
                  class="button-black products__image__caption"
                  href="/catalogue?categories=men"
                >
                  for woman
                </a>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div
        class="products__wrapper"
      >
        <div
          class="container products"
        >
          <div
            class="products__images"
          >
            <div
              class="products__images"
            >
              <img
                alt=""
                src="/Images/forWomen.jpg"
              />
              <a
                class="button-black products__image__caption"
                href="/catalogue?categories=woman"
              >
                for woman
              </a>
            </div>
            <div
              class="products__images"
            >
              <img
                alt=""
                src="/Images/forKids.jpg"
              />
              <a
                class="button-black products__image__caption"
                href="/catalogue?categories=children"
              >
                for woman
              </a>
            </div>
            <div
              class="products__images"
            >
              <img
                alt=""
                src="/Images/forMen.jpg"
              />
              <a
                class="button-black products__image__caption"
                href="/catalogue?categories=men"
              >
                for woman
              </a>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
